system/jlib/jtrace.cpp

```diff
@@ -69,3 +83,282 @@ const char* LogTrace::queryLocalId() const
+            auto exporter = opentelemetry::exporter::trace::OStreamSpanExporterFactory::Create();
+
+            //SimpleSpanProcesser sends spans one by one to an exporter.
+            //We could use a batchspanprocessor, which will group several spans together, before sending them to an exporter.
+            auto processor = opentelemetry::sdk::trace::SimpleSpanProcessorFactory::Create(std::move(exporter));
+            std::vector<std::unique_ptr<opentelemetry::sdk::trace::SpanProcessor>> processors;
+            processors.push_back(std::move(processor));
+
+            // Default is an always-on sampler.
+            std::shared_ptr<opentelemetry::sdk::trace::TracerContext> context =
+                opentelemetry::sdk::trace::TracerContextFactory::Create(std::move(processors));
+            std::shared_ptr<opentelemetry::trace::TracerProvider> provider =
+                opentelemetry::sdk::trace::TracerProviderFactory::Create(context);
+
+            // Set the global trace provider
+            opentelemetry::trace::Provider::SetTracerProvider(provider);
+
+            // set global propagator
+            // Injects Context into and extracts it from carriers that travel in-band
+            // across process boundaries. Encoding is expected to conform to the HTTP
+            // Header Field semantics.
+            // Values are often encoded as RPC/HTTP request headers.
+            opentelemetry::context::propagation::GlobalTextMapPropagator::SetGlobalPropagator(
+                opentelemetry::nostd::shared_ptr<opentelemetry::context::propagation::TextMapPropagator>(
+                    new opentelemetry::trace::propagation::HttpTraceContext()));
+        }
+    }
+    catch (IException * e)
+    {
+        EXCLOG(e);
+        e->Release();
+    }
+}
+
+void CTraceManager::cleanupTracer()
+{
+    std::shared_ptr<opentelemetry::trace::TracerProvider> none;
+    opentelemetry::trace::Provider::SetTracerProvider(none);
+}
+
+void CSpan::setAttributes(const IProperties * attributes)
+{
+    Owned<IPropertyIterator> iter = attributes->getIterator();
+    ForEach(*iter)
+    {
+        const char * key = iter->getPropKey();
+        const char * val = attributes->queryProp(key);
+        span->SetAttribute(key, val);
+    }
+}
+
+void CTransactionSpan::setAttriburesFromHTTPHeaders(StringArray & httpHeaders)
+{
+    ForEachItemIn(currentHeaderIndex, httpHeaders)
+    {
+        const char* httHeader = httpHeaders.item(currentHeaderIndex);
+        if(!httHeader)
+            continue;
+
+        if(stricmp(httHeader, HPCCSemanticConventions::kGLOBALIDHTTPHeader) == 0)
+        {
+            hpccGlobalId.set(httpHeaders.item(currentHeaderIndex+1));
+            continue;
+        }
+
+        if(stricmp(httHeader, HPCCSemanticConventions::kCallerIdHTTPHeader) == 0)
+        {
+            hpccCallerId.set(httpHeaders.item(currentHeaderIndex+1));
+            continue;
+        }
+    }
```

Comment:
I would expect this function and the next to do the same thing.  This doesn't contain the latter code.

For simplicitly I would create an IProperties from the array and then call the other function.  If it is time critical then investigate some more later.